File: pkg/export/markdown/options.go
// pkg/export/markdown/options.go
package markdown

import (
	"fmt"
	"strings"

	"github.com/sruja-ai/sruja/pkg/export/mermaid"
)

// ContextType defines the context for export formatting
type ContextType string

const (
	ContextDefault        ContextType = "default"
	ContextCodeGeneration ContextType = "code_generation"
	ContextReview         ContextType = "review"
	ContextAnalysis       ContextType = "analysis"
)

// Scope defines what part of the architecture to export
type Scope struct {
	Type string // "system", "container", "component", "full"
	ID   string // Element ID (e.g., "OrderService")
}

// ParseScope parses a scope string like "system:OrderService"
func ParseScope(scopeStr string) (*Scope, error) {
	if scopeStr == "" || scopeStr == "full" {
		return &Scope{Type: "full", ID: ""}, nil
	}

	parts := strings.SplitN(scopeStr, ":", 2)
	if len(parts) != 2 {
		return nil, fmt.Errorf("invalid scope format: %s (expected 'type:id')", scopeStr)
	}

	scopeType := strings.ToLower(strings.TrimSpace(parts[0]))
	scopeID := strings.TrimSpace(parts[1])

	switch scopeType {
	case "system", "container", "component":
		return &Scope{Type: scopeType, ID: scopeID}, nil
	default:
		return nil, fmt.Errorf("invalid scope type: %s (expected system, container, or component)", scopeType)
	}
}

// Options represents Markdown export options.
type Options struct {
	IncludeTOC          bool
	IncludeOverview     bool
	IncludeSystems      bool
	IncludeDeployments  bool
	IncludePersons      bool
	IncludeRequirements bool
	IncludeADRs         bool
	IncludeScenarios    bool
	IncludeMetadata     bool
	IncludeGlossary     bool
	MermaidConfig       mermaid.Config
	HeadingLevel        int

	// New AI-friendly options
	Scope      *Scope      // Scope to specific element
	TokenLimit int         // Maximum tokens (0 = no limit)
	Context    ContextType // Context type for formatting
}

// DefaultOptions returns the default Markdown export options.
func DefaultOptions() Options {
	return Options{
		IncludeTOC:          true,
		IncludeOverview:     true,
		IncludeSystems:      true,
		IncludeDeployments:  true,
		IncludePersons:      true,
		IncludeRequirements: true,
		IncludeADRs:         true,
		IncludeScenarios:    true,
		IncludeMetadata:     true,
		IncludeGlossary:     true,
		MermaidConfig:       mermaid.DefaultConfig(),
		HeadingLevel:        1,
		Scope:               &Scope{Type: "full", ID: ""},
		TokenLimit:          0,
		Context:             ContextDefault,
	}
}

File: crates/sruja-export/src/markdown/options.rs
//! Markdown export options (ported from Go)

use crate::mermaid::exporter::MermaidConfig;

#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
pub enum ContextType {
    #[default]
    Default,
    CodeGeneration,
    Review,
    Analysis,
}

#[derive(Debug, Clone)]
pub struct Scope {
    pub r#type: String, // "system", "container", "component", "full"
    pub id: String,
}

impl Scope {
    pub fn full() -> Self {
        Self {
            r#type: "full".to_string(),
            id: String::new(),
        }
    }
}

#[derive(Debug, Clone)]
pub struct MarkdownOptions {
    /// Top-level document title (e.g. "Architecture Overview"). When None, derived from overview summary or default.
    pub document_title: Option<String>,
    pub include_toc: bool,
    pub include_overview: bool,
    pub include_mermaid_diagrams: bool,
    pub include_systems: bool,
    pub include_deployments: bool,
    pub include_persons: bool,
    pub include_requirements: bool,
    pub include_adrs: bool,
    pub include_scenarios: bool,
    pub include_metadata: bool,
    pub include_glossary: bool,
    pub include_recommendations: bool,
    /// When true, add a "Relations" section listing all relations (from → to "label").
    pub include_relations: bool,
    pub mermaid_config: MermaidConfig,
    pub heading_level: u32,
    pub scope: Scope,
    pub token_limit: usize,
    pub context: ContextType,
    pub use_views: bool,
    pub view_name: Option<String>,
    /// When true and use_views is true (and view_name is None), emit a "Custom views" section with all defined views.
    pub include_all_views: bool,
}

impl Default for MarkdownOptions {
    fn default() -> Self {
        Self {
            document_title: None,
            include_toc: true,
            include_overview: true,
            include_mermaid_diagrams: true,
            include_systems: true,
            include_deployments: true,
            include_persons: true,
            include_requirements: true,
            include_adrs: true,
            include_scenarios: true,
            include_metadata: true,
            include_glossary: true,
            include_recommendations: true,
            include_relations: false,
            mermaid_config: MermaidConfig::default(),
            heading_level: 1,
            scope: Scope::full(),
            token_limit: 0,
            context: ContextType::Default,
            use_views: false,
            view_name: None,
            include_all_views: false,
        }
    }
}
